src/main/java/DilemmaDetector/ScenarioReader.java

```diff
@@ -170,8 +174,17 @@ private Driver getDriverFromOntology(int number) {
         ArrayList<Surrounding> left_surrounding = new ArrayList<>();
         ArrayList<Surrounding> right_surrounding = new ArrayList<>();
 
-        left_surrounding.addAll(scenario.getHas_surrounding_left());
-        right_surrounding.addAll(scenario.getHas_surrounding_right());
+        for (Surrounding s: scenario.getHas_surrounding_left()){
+            String surroundingName = s.getOwlIndividual().getIRI().toString();
+            s = getSurroundingAsSpecificClass(surroundingName);
+            left_surrounding.add(s);
+        }
+
+        for (Surrounding s: scenario.getHas_surrounding_right()){
+            String surroundingName = s.getOwlIndividual().getIRI().toString();
+            s = getSurroundingAsSpecificClass(surroundingName);
+            right_surrounding.add(s);
+        }
```

Comment:
Tu jest właściwie powtórzenie dwóch takich samych forów, można do osobnej funkcji wyciągnąć